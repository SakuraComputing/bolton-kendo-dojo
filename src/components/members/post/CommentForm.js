import React, { Component } from 'react'

class commentForm extends Component {
    render() {
        return (
            <div>
                
            </div>
        )
    }
}
export default commentForm;
