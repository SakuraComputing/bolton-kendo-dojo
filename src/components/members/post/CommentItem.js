import React, { Component } from 'react';
import { connect } from 'react-redux';
import { deleteComment } from '../../../actions/postActions';

export class CommentItem extends Component {

    onDeleteClick(postId, commentId) {
        this.props.deleteComment(postId, commentId);
    }

    render() {
        const { comment, postId, auth } = this.props;

        return(
            <div className="post-container">
                <div className="comment-outer">
                    <div className="comment-inner-1">
                        <a href="profile.html">
                            <img className="rounded-circle d-none d-md-block"
                                    src={comment.avatar} alt=""/>
                        </a>
                        <br/>
                        <p className="avatar-name">{comment.name}</p>
                        <div className="comment-container">
                        </div>
                    </div>
                    <div className="comment-inner-2">
                        <p className="comment-text">{comment.text}</p>
                    </div>
                    <div className="comment-inner-3">
                        {comment.user === auth.user.id ? (
                            <button onClick={this.onDeleteClick.bind(this, postId, comment._id)}
                                    type="button"
                                    className="button-small button-delete item-delete"
                            >
                            <i className="fas fa-times"></i>
                            Delete Comment
                            </button>
                        ) : null}
                    </div>
                </div>
            </div>
        )
    }
}

const mapStateToProps = (state) => ({
    auth: state.auth
});
 
export default connect(mapStateToProps, {deleteComment})(CommentItem);
