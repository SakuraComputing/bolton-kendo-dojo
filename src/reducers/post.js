import {
    GET_POSTS,
    ADD_POST,
    GET_ERRORS,
    DELETE_POST,
    GET_POST

} from '../actions/types';

const postReducerDefaultState = [];

export default (state = postReducerDefaultState, action) => {
    switch(action.type) {
        
    }
}
