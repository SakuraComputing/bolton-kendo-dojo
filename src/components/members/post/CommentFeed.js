import React, { Component } from 'react';
import CommentItem from './CommentItem';

class commentFeed extends Component {
    render() {
        const {comments, postId} = this.props;

        return comments.map(comment => (
            <CommentItem key={comment._id} comment={comment} postId={postId}/>
        ))
    }
}
export default commentFeed;
