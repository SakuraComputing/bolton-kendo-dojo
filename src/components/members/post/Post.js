import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { getPost } from '../../../actions/postActions';
import Spinner from '../../../common/Spinner';
import PostItem from '../posts/PostItem';
import CommentForm from './CommentForm';
import CommentFeed from './CommentFeed';

export class Post extends Component {

    componentDidMount() {
        this.props.getPost(this.props.match.params.id);
    }

    render() {
        const { post, loading } = this.props.post;

        let postContent;

        if(post === null || loading || Object.keys(post).length === 0) {
            postContent = <Spinner />
        } else {
            postContent = (
                <div>
                    <PostItem post={post} showActions={false}/>
                    <CommentForm postId={post._id}/>
                    <CommentFeed postId={post._id} comments={post.comments} />
                </div>
            )
        }
        
        return (
            <div>
                <div className="back-to-feed">
                    <Link to="/feed" className="button-small button-warning">
                        Back To Feed
                    </Link>
                </div>
                {postContent}
            </div>
        )
    }
}
const mapStateToProps = state => ({
    post: state.post
})
export default connect(mapStateToProps, { getPost })(Post);
