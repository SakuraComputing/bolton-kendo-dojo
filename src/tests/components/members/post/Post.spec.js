import React from 'react';
import { shallow } from 'enzyme';
import { Post } from '../../../../components/members/post/Post';

describe('Comment Feed Component', () => {
  let post, props;

  beforeEach(() => {
      props = {
            post: {
                post: jest.fn(),
                loading: false
            },
            match: {
                params: jest.fn()
            },
            getPost: jest.fn()
        };
        post = shallow(<Post {...props}/>);
  });
  it('should render correctly', () => {
      expect(post).toMatchSnapshot();
  });
})
