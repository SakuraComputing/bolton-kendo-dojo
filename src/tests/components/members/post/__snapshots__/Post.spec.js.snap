// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Feed Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Post
    getPost={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    match={
      Object {
        "params": [MockFunction],
      }
    }
    post={
      Object {
        "loading": false,
        "post": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="back-to-feed"
        >
          <Link
            className="button-small button-warning"
            replace={false}
            to="/feed"
          >
            Back To Feed
          </Link>
        </div>,
        <div>
          <Connect(PostItem)
            post={[MockFunction]}
            showActions={false}
          />
          <Connect(CommentForm) />
          <commentFeed />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            className="button-small button-warning"
            replace={false}
            to="/feed"
          >
            Back To Feed
          </Link>,
          "className": "back-to-feed",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Back To Feed",
            "className": "button-small button-warning",
            "replace": false,
            "to": "/feed",
          },
          "ref": null,
          "rendered": "Back To Feed",
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(PostItem)
              post={[MockFunction]}
              showActions={false}
            />,
            <Connect(CommentForm) />,
            <commentFeed />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "post": [MockFunction],
              "showActions": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "postId": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "comments": undefined,
              "postId": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="back-to-feed"
          >
            <Link
              className="button-small button-warning"
              replace={false}
              to="/feed"
            >
              Back To Feed
            </Link>
          </div>,
          <div>
            <Connect(PostItem)
              post={[MockFunction]}
              showActions={false}
            />
            <Connect(CommentForm) />
            <commentFeed />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button-small button-warning"
              replace={false}
              to="/feed"
            >
              Back To Feed
            </Link>,
            "className": "back-to-feed",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Back To Feed",
              "className": "button-small button-warning",
              "replace": false,
              "to": "/feed",
            },
            "ref": null,
            "rendered": "Back To Feed",
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(PostItem)
                post={[MockFunction]}
                showActions={false}
              />,
              <Connect(CommentForm) />,
              <commentFeed />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "post": [MockFunction],
                "showActions": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "postId": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "comments": undefined,
                "postId": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
